pkgs/development/python-modules/tinygrad/default.nix

```diff
@@ -175,6 +183,10 @@ buildPythonPackage rec {
     "extra/"
   ];
 
+  passthru.tests = {
+    withCuda = tinygrad.override { cudaSupport = true; };
```

Comment:
how is this used in nixpkgs, given ofborg will never run this ? ( due to licensing )

Reply:
It lets you run `nix-build -A python312Packages.tinygrad.tests` to check if the package builds and runs fine on CUDA.
I find that more convenient than having to manually building the package for CUDA, let alone thinking about doing it.

Reply:
I can see the convenience for maintainer, also don't you also need `allowUnfree` ?

Reply:
You need to have `allowUnfree` in your local nixpkgs configuration indeed.

Reply:
since `allowUnfree` is not default for nixpkgs, is this a usefull change ( this is a question for everyone) ? eg ofborg will not eval this.

Reply:
it will be the same as wenn ofborg tries to check an unfree package.